### ISSUE TITLE
Snoei: a.k.a. how to translate technical jargon

### ISSUE THREAD

**Issue by Sjors** [bitcoin/bitcoin#11875]
This is hilarious as it sounds like gardening advice.
<img width="432" alt="snoei" src="https://user-images.githubusercontent.com/10217/33882565-cd7c1450-df38-11e7-9261-64c6fd5f3ee0.png">

I suggest sticking to the English technical terms, appending a translation in brackets where needed. For custom settings like `prune=`, it's safe to assume the user knows the English word.

**Comment by fanquake** [comment]
Please report translation issues upstream to Transifex.

**Comment by Sjors** [comment]
Just joined the Transifex team. I'll flag this specific issue there. I'm getting a "There was an error connecting to server." when trying to view the strings, so will try that again later.

However, I'd also like to change the [translation strings policy](https://github.com/bitcoin/bitcoin/blob/master/doc/translation_strings_policy.md) to reflect this, if people think that's a good idea.

I see there's no review for the past few Dutch releases:
<img width="991" alt="schermafbeelding 2017-12-12 om 12 49 06" src="https://user-images.githubusercontent.com/10217/33883122-de858694-df3a-11e7-8981-67dba3a26be9.png">

We should consider not merging unreviewed translations. Not sure what the fallback behavior should be; use the previous reviewed translation if a string is unchanged, fall back to English if a string has changed but the translation is unreviewed?

**Comment by sipa** [comment]
There is a meta issue here that perhaps is more appropriate to discuss here.

How do we translate jargon? The answer is perhaps language and culturally dependent. I can't imagine that anyone is served by seeing terms like "snoei" and "blokketen" (literal translation of blockchain).

**Comment by jonasschnelli** [comment]
Maybe we should define a list of words that should not be translated?
The same problem appears in German, where words like "wallet", "blockchain" and even "HD" (for hd wallet) is translated which makes little sense to me.

**Comment by Sjors** [comment]
One approach that has worked for me in the past when dealing with application translations, is to have translator instructions. There should be universal instructions and language specific instructions. That can probably go on one page, e.g. on the docs repo, with a link form Transifex, so that translators see it.

We can define a list at @jonasschnelli suggested and add that to the universal instructions, but the list should merely be a guidelines. For instance, Mandarin does have its own [word for Bitcoin](https://www.youtube.com/watch?v=pADgAmNzxek).

**Comment by MarcoFalke** [comment]
A workaround, to forbid translating a specific piece of text, is to exclude it from the translation string.

That makes sense for command line args, such as `-prune`, but might drain the context for general words such as "wallet". Thus, make translations harder.

I think there is a Glossary on transifex, where a noun and its translation can be put.

![screenshot from 2017-12-12 20-09-17](https://user-images.githubusercontent.com/6399679/33916761-758a3408-df78-11e7-8365-33906dd8a2cc.png)

**Comment by Sjors** [comment]
Using the glossary feature sounds like the better approach, as long as reviewers also check it. Transifex shows the glossary for any matching words of the phrase the translator is working on, so that's quite nice.

It's already possible to only download reviewed translations:

<img width="382" alt="schermafbeelding 2017-12-13 om 09 14 56" src="https://user-images.githubusercontent.com/10217/33928503-65bfaf3a-dfe6-11e7-935e-4547c840c480.png">

The word "snoei" does not appear in the reviewed file, only in the unreviewed file.

Perhaps the automated script can be modified to use the reviewed file instead? This might cause a lot of strings to get dropped, so we should try to get most languages reviewed before making such a change.

**Comment by laanwj** [comment]
Of foremost importance is that we don't flag overly technical messages for translation. This is also why I opened #10962.

**Comment by Sjors** [comment]
I'll take a look at the script.

Also [link to recent IRC discussion](https://botbot.me/freenode/bitcoin-core-dev/2017-12-13/?msg=94594087&page=1).

The error message I was getting appears to have been a permission issue in disguise.

**Comment by Sjors** [comment]
`tx pull -f -a --mode=reviewed`should [do the trick](https://docs.transifex.com/client/pull).

In a new directory I did:

```
tx set --auto-remote https://www.transifex.com/bitcoin/bitcoin/qt-translation-015x
tx pull -f -l nl
```

This will have "snoei" in it.

Then:
```
tx pull -f -l nl --mode=reviewed
```
This won't have "snoei" in it. In fact, it will have no translations because none of the phrases have been reviewed in Transifex.

If you use `-l de` you'll get the German translations that have been reviewed, e.g. compare "Transaction amounts must not be negative" with and without `--reviewed`.

When I change the `update-translations.py` script, all the unreviewed translation are removed as expected.

```python
subprocess.call([TX, 'pull', '-f', '-a', '--mode=reviewed'])
```

This won't if you call it for just a single language, just FYI:

```python
subprocess.call([TX, 'pull', '-f', '--mode=reviewed', '-l de'])
```

`src/qt/de.ts` doesn't change when I do that.

**Comment by Sjors** [comment]
Also every language has a team that people can join. It's just that the join button might be broken, in which case support will add you if you contact them.

**Comment by Sjors** [comment]
Once you figure out your way through the SAAS Kafka, Transifex is pretty powerful.

For example it will warn you if you don't use the translation suggested in the glossary. E.g. when I use "wallet"* instead of "portefuille":

<img width="712" alt="schermafbeelding 2017-12-13 om 22 06 25" src="https://user-images.githubusercontent.com/10217/33962466-ec115e9c-e051-11e7-93ab-012b13839d5e.png">

<img width="401" alt="schermafbeelding 2017-12-13 om 22 06 38" src="https://user-images.githubusercontent.com/10217/33962469-ed8a02f6-e051-11e7-82ca-28bcd42f5387.png">

* = I found several Dutch state television news articles that use the English term "wallet", but I only changed it here to illustrate the feature.
